api/games/methods.js

```diff
@@ -60,3 +62,58 @@ export const updateGameData = new ValidatedMethod({
     }
   }
 });
+
+export const earlyExitGame = new ValidatedMethod({
+  name: "Games.methods.earlyExitGame",
+
+  validate: new SimpleSchema({
+    gameId: {
+      type: String,
+      regEx: SimpleSchema.RegEx.Id
+    },
+    endReason: {
+      label: "Reason for End",
+      type: String,
+      regEx: /[a-zA-Z0-9_]+/
+    }
+  }).validator(),
+
+  run({ gameId, endReason }) {
+    if (!Meteor.isServer) {
+      return;
+    }
+
+    const game = Games.findOne(gameId);
+    if (!game) {
+      throw new Error("game not found");
+    }
+
```

Comment:
I think we should be checking whether the game has already ended here. Normally this shouldn't happen twice obviously, but just to protect from a "double-click" type symptom. Not sure what issue might be with doing it twice, other than changing the timestamp. Should probably log the subsequent calls too, so in dev, experimenters can detect code calling game.end() multiple times.